Xamarin.Forms.Platform.iOS/Renderers/ListViewRenderer.cs

```diff
@@ -654,6 +657,18 @@ void UpdateSeparatorVisibility()
 			}
 		}
 
+		void UpdateSelectionMode()
+		{
+			if (Element.SelectionMode == ListViewSelectionMode.None)
+			{
+				Element.SelectedItem = null;
```

Comment:
For example here you are not setting the index to -1

Reply:
There's no `SelectedIndex` on iOS. The equivalent is done in the next few rows (lines 665-667). Please see my previous comment as well.